hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/SendEventOperation.java

```diff
@@ -41,7 +41,10 @@ public SendEventOperation(EventEnvelope eventEnvelope, int orderKey) {
     @Override
     public void run() throws Exception {
         EventServiceImpl eventService = (EventServiceImpl) getNodeEngine().getEventService();
-        eventService.executeEventCallback(new EventProcessor(eventService, eventEnvelope, orderKey));
+        Registration registration = eventService.getRegistration(eventEnvelope);
```

Comment:
`eventService.getRegistration()` assumes registrations are always local. I don't fully understand the flow, but it's somewhat suspicious.